frappe/desk/doctype/todo/test_todo.py

```diff
@@ -47,6 +49,30 @@ def test_fetch_setup(self):
 		self.assertEqual(todo.assigned_by_full_name,
 			frappe.db.get_value('User', todo.assigned_by, 'full_name'))
 
+	def test_access(self):
+		todo1 = create_new_todo('Test1', 'Administrator')
+		todo2 = create_new_todo('Test2', 'test4@example.com')
+
+		frappe.set_user('test4@example.com')
+		test_user_data = DatabaseQuery('ToDo').execute()
+
+		self.assertFalse(todo1.has_permission("read"))
+		self.assertFalse(todo1.has_permission("write"))
+		self.assertTrue(todo2.has_permission("read"))
+		self.assertTrue(todo2.has_permission("write"))
+
+		frappe.set_user('Administrator')
+		admin_data = DatabaseQuery('ToDo').execute()
+
+		self.assertTrue(todo1.has_permission("read"))
+		self.assertTrue(todo1.has_permission("write"))
+		self.assertTrue(todo2.has_permission("read"))
+		self.assertTrue(todo2.has_permission("write"))
+
+		self.assertNotEqual(test_user_data, admin_data)
+
+		frappe.db.rollback()
+
```

Comment:
A test case to check only read access via role permission is missing.

Reply:
@surajshetty3416 Only to check if the user or admin has read access to ToDo or not, right?

Reply:
To check whether a user (who is not assigned_by or owner) with a specific role is able to access the ToDo document or not